### PR TITLE
[openrec] add support for openrec.tv

### PR DESCRIPTION
**Pull request by Lesmiscore**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

Note that the code is taken from [my fork](https://github.com/ytdl-patched/ytdl-patched/blob/ytdlp/yt_dlp/extractor/openrec.py), which is my own work.

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This PR adds support for openrec.tv, which is like Twitch but in Japan. Not geo-restricted I think.

This does not support replay-from-start feature for live, because I don't have Premium subscription.

example URL for live: https://www.openrec.tv/live/kdr7xq0l38j (running as of writing, but you can find new one at https://www.openrec.tv/popular/live )
example URL for capture (so-called "clip" in Twitch or so): https://www.openrec.tv/capture/kvn4eorkzqe

### CODE REVIEW

**Comment by nao20010128nao** [comment]
memo: voicy and mirrativ

File: yt_dlp/extractor/openrec.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        channel_user = movie_store.get('channel', {}).get('user')
+        if channel_user:
+            uploader = channel_user['name']
+            uploader_id = channel_user['id']
+        else:
+            self.report_warning('Failed to extract uploader information')
+            uploader = channel_user.get('name')
+            uploader_id = channel_user.get('id')
+
+        timestamp = movie_store.get('startedAt', {}).get('time')
```
Comment (pukkandan) [inline yt_dlp/extractor/openrec.py:48]: Use try_get / traverse_obj

File: yt_dlp/extractor/openrec.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        thumbnail = movie_store.get('thumbnailUrl')
+
+        channel_user = movie_store.get('channel', {}).get('user')
+        if channel_user:
+            uploader = channel_user['name']
+            uploader_id = channel_user['id']
+        else:
+            self.report_warning('Failed to extract uploader information')
+            uploader = channel_user.get('name')
+            uploader_id = channel_user.get('id')
```
Comment (pukkandan) [inline yt_dlp/extractor/openrec.py:46]: Why so complicated?
Isnt this the same as L106-108?

File: yt_dlp/extractor/openrec.py (hunk 1 above)
Comment (nao20010128nao) [reply yt_dlp/extractor/openrec.py:48]: I didn't notice while refactoring the code, will fix

File: yt_dlp/extractor/openrec.py (hunk 2 above)
Comment (nao20010128nao) [reply yt_dlp/extractor/openrec.py:46]: I didn't notice while refactoring the code, will fix

File: yt_dlp/extractor/openrec.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    try_get,
+    unified_strdate
+)
+from ..compat import compat_str
+
+
+class OpenRecBaseIE(InfoExtractor):
+    pass
+
+
```
Comment (pukkandan) [inline yt_dlp/extractor/openrec.py:17]: Why?

Comment (nao20010128nao) [reply yt_dlp/extractor/openrec.py:17]: idk, but definitely unneeded

**Pull request merged**